indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -731,7 +731,8 @@ public void onFailure(Throwable t)
           }
 
           if (System.currentTimeMillis() > nextCheckpointTime) {
-            sequenceToCheckpoint = getLastSequenceMetadata();
+            var lastSequenceMetadata = getLastSequenceMetadata();
```

Comment:
Change to 
`SequenceMetadata<PartitionIdType, SequenceOffsetType> lastSequenceMetadata = getLastSequenceMetadata();`

Reply:
See Herb Sutter's arguments on `auto` the same apply here.

Reply:
See Herb Sutter's arguments on `auto` in this video: https://youtu.be/xnqTKD8uD64?t=1703
While he talks about this in context of C++, the same arguments apply here.

Reply:
The semaphore build was failing due to "var"